movies/tests/test_movie_api_utils.py

```diff
@@ -30,3 +30,20 @@ def test_movies_now_playing_success(self):
 
         for movie in movies["results"]:
             self.assertTrue(movie)
+
+    def test_bad_json_in_date_conversion(self):
+        movies = self.movie_api_utils.MOVIES_IN_THEATERS
+        with self.assertRaisesMessage(
+            MovieApiException, "Ensure JSON argument is a dictionary object."
+        ):
+            self.movie_api_utils.convert_date_string_into_object(
+                movies, filter="results"
+            )
+
+    def test_bad_filter_value_in_date_conversion(self):
+        movies = self.movie_api_utils.get_movies_now_playing()
+        filter = "bad_filter"
+        with self.assertRaisesMessage(
+            MovieApiException, f"{filter} is not a valid attribute on a movie object."
+        ):
+            self.movie_api_utils.convert_date_string_into_object(movies, filter=filter)
```

Comment:
Include a test for a good filter.
- happy-path tests prove that what you intended works and allows for other developers to quickly understand the code by simply reviewing the tests.